domains/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -216,84 +216,50 @@ pub mod pallet {
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        /// Updates the light client state
         #[pallet::weight(T::WeightInfo::sync_committee_period_update())]
         #[transactional]
-        pub fn sync_committee_period_update(
+        #[pallet::call_index(0)]
+        pub fn light_client_update(
             origin: OriginFor<T>,
-            sync_committee_period_update: SyncCommitteePeriodUpdateOf<T>,
+            light_client_update: LightClientUpdateOf,
         ) -> DispatchResult {
             let _sender = ensure_signed(origin)?;
 
             Self::check_bridge_blocked_state()?;
 
-            let sync_committee_period = sync_committee_period_update.sync_committee_period;
+            let sync_committee_period = light_client_update.sync_committee_period;
             log::info!(
                 target: "ethereum-beacon-client",
-                "💫 Received sync committee update for period {}. Applying update",
+                "💫 Received light client update for sync committee period {}. Applying update",
                 sync_committee_period
             );
 
-            if let Err(err) =
-                Self::process_sync_committee_period_update(sync_committee_period_update)
-            {
+            Self::process_light_client_update(light_client_update).map_err(|err| {
                 log::error!(
                     target: "ethereum-beacon-client",
-                    "💫 Sync committee period update failed with error {:?}",
+                    "💫 Light client update failed with error {:?}",
```

Comment:
Not sure I understand this either. Why the refactor ?

Reply:
[See my comment](https://github.com/subspace/subspace/pull/1170#issuecomment-1453590351)